tests/Unit/API/Google/AdsCampaignTest.php

```diff
@@ -109,6 +114,10 @@ public function test_get_campaigns() {
 		];
 
 		$this->generate_ads_campaign_query_mock( $campaigns_data, $campaign_criterion_data );
+
+		$this->service_client->expects( $this->exactly( 2 ) )
+			->method( 'search' );
```

Comment:
`service_client` is kind of part of the HelperTrait, so I think it would be better to move this expectation within the `generate_ads_campaign_query_mock` since that's the function that returns two results. Otherwise it looks a bit confusing what is expected in this test, and why it's not expected in the tests `test_get_campaigns_with_nonexist_location_id` and `test_get_campaigns_with_invalid_location_id`.

Reply:
Also maybe we can update the doc block for `generate_ads_campaign_query_mock` since it now generates mocked responses for two queries.

Reply:
Thanks, good suggestions! Updated in 40fafee.